src/app/core/layers/layersstore.js

```diff
@@ -112,12 +112,14 @@ proto.getLayersDict = function(filter = {}, options = {}) {
     layers = layers.filter(layer => -1 !== ids.indexOf(layer.getId()));
   }
 
-  // return only selected if some one are selected
+  //in case of SELECTED_OR_ALL = true
   if (filter.SELECTED_OR_ALL) {
     let _layers = layers;
+    //check if there are selected layer
     layers = layers.filter(layer => layer.isSelected());
-    layers = layers.length ? layers : _layers;
-  } else if (false === filter.SELECTED_OR_ALL) {
+    //otherwise get all layers
+    layers = layers.length > 0 ? layers : _layers;
+  } else if ('boolean' === typeof filter.SELECTED) { //check if filter.SELECTED is Boolean
```

Comment:
@volterra79 so what are the possible values for `filter.SELECTED`?

Reply:
@Raruto can be undefined or null. It depends if is set or not

Reply:
@Raruto these changes solve also a issue related to add feature to result button

![Screenshot from 2024-02-06 12-27-04](https://github.com/g3w-suite/g3w-client/assets/1051694/c7c193dc-8fd5-4b14-990f-246387716c7b)